api/envoy/extensions/http/stateful_session/cookie/v3/cookie.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package envoy.extensions.http.stateful_session.cookie.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.stateful_session.cookie.v3";
+option java_outer_classname = "CookieProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Cookie based stateful session extension]
+
+// This extension allows the session state to be tracked via cookies.
+//
+// This extension first encodes the address of the upstream host selected by the load
+// balancer into a `set-cookie` response header with the configured name.
+// This extension will also try to parse the specific upstream host address from
+// the named cookie when new requests come. If the address parsed from the cookie
+// corresponds to a valid upstream host, this upstream host will be selected first.
+// See :ref:`stateful session filter
+// <envoy_v3_api_msg_extensions.filters.http.stateful_session.v3.StatefulSession>`.
+//
+// [#extension: envoy.http.stateful_session.cookie]
+message CookieBasedSessionState {
+  // The name of the cookie that will be used to store the encoded
+  // upstream host.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // If specified, a TTL will be added to new session state
+  // cookies in the set-cookie header. Specify a TTL of 0 to
```

Comment:
Nit. Set this to `0` to use a session cookie.

Does this mean we use whatever is already there?

Reply:
I am not sure I get it. `0` TTL means that no `Max-Age` for the cookie.